datafusion/common/src/utils/mod.rs

```diff
@@ -746,6 +746,27 @@ pub fn combine_limit(
     (combined_skip, combined_fetch)
 }
 
+pub fn get_map_entry_field(data_type: &DataType) -> Result<&Fields> {
+    match data_type {
+        DataType::Map(filed, _) => {
+            let field_data_type = filed.data_type();
+            match field_data_type {
+                DataType::Struct(fields) if fields.len() == 2 => Ok(fields),
+                DataType::Struct(fields) => {
+                    _internal_err!(
+                        "Expected a Struct type with 2 fields in Map, got {} fields",
+                        fields.len()
+                    )
+                }
+                _ => {
+                    _internal_err!("Expected a Struct type, got {:?}", field_data_type)
+                }
+            }
+        }
```

Comment:
Do we really need all these checks ? Does the data type being Map ensure that all these conditions are alread checked ?